File: cs50x/sentimental-credit/credit.py
# TODO
num = input("Number: ")
sum1 = 0
if len(num) not in [13, 15, 16]:
    print("INVALID")
    exit()
if (int(num[0:2]) not in [34, 37, 51, 52, 53, 54, 55] and int(num[0]) != 4):
    print("INVALID")
    exit()
temp = num[-2::-2]
for i in temp:
    sum1 += int(i)*2 if int(i)*2 < 10 else sum(int(x) for x in str(int(i)*2))
temp = num[-1::-2]
for i in temp:
    sum1 += int(i)
if str(sum1)[-1] != "0":
    print(str(sum1)[-1])
    print("INVALID")
    exit()
array = [3, 4, 5]
type = array.index(int(num[0]))
if type == 0:
    print("AMEX")
elif type == 1:
    print("VISA")
elif type == 2:
    print("MASTERCARD")

File: cs50x/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long cc = get_long("Number: ");
    int sum = 0;
    int length = 0;
    int digits[16];
    while (cc != 0)
    {
        digits[length] = cc % 10;
        cc /= 10;
        length++;
    }
    for (int i = 0; i < length; i++)
    {
        if (i % 2 != 0)
        {
            if (2 * (digits[i]) >= 10)
            {
                int temp = 2 * (digits[i]);
                while (temp != 0)
                {
                    sum += temp % 10;
                    temp /= 10;
                }
            }
            else
            {
                sum += 2 * (digits[i]);
            }
        }
        else
        {
            sum += digits[i];
        }
    }

    if (sum % 10 == 0 && length > 12)
    {
        if (digits[length - 1] == 3 && (digits[length - 2] == 4 || digits[length - 2] == 7))
        {
            printf("AMEX\n");
        }
        else if (digits[length - 1] == 4)
        {
            printf("VISA\n");
        }
        else if (digits[length - 1] == 5 && (digits[length - 2] > 0 && digits[length - 2] < 6))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}
